src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -16,222 +16,75 @@
-	 * @param f the features to be processed, must be of type
-	 * DenseFeatures<float64_t>
-	 * @return true if new random coefficients were generated, false if old ones
-	 * from a call to set_randomcoefficients(...) are kept
-	 */
-	virtual void fit(std::shared_ptr<Features> f);
-
-	/**  setter for kernel width
-	 * @param width kernel width to be set
-	 */
-	void set_kernelwidth(const float64_t width);
-
-	/**  getter for kernel width
-	 * @return kernel width
-	 * throws exception if kernelwidth <=0
-	 */
-	float64_t get_kernelwidth( ) const;
-
-	/**  getter for the random coefficients
-	 * necessary for creating random fourier features compatible to the current ones
-	 * returns values of internal members randomcoeff_additive
-	 * and randomcoeff_multiplicative
-	 */
-	void get_randomcoefficients(float64_t ** randomcoeff_additive2,
-			float64_t ** randomcoeff_multiplicative2,
-			int32_t *dim_feature_space2, int32_t *dim_input_space2, float64_t* kernelwidth2 ) const;
-
-	/**  setter for the random coefficients
-	 * necessary for creating random fourier features compatible to the previous ones
-	 * sets values of internal members	randomcoeff_additive
-	 * and randomcoeff_multiplicative
-	 * simply use as input what you got from get_random_coefficients(...)
-	 */
-	void set_randomcoefficients(float64_t *randomcoeff_additive2,
-			float64_t * randomcoeff_multiplicative2,
-			const int32_t dim_feature_space2, const int32_t dim_input_space2, const float64_t kernelwidth2);
-
-	/** a setter
-	 * @param dim the value of protected member dim_input_space
-	 * throws a shogun exception if dim<=0
-	 */
-	void set_dim_input_space(const int32_t dim);
-
-	/** a setter
-	 * @param dim the value of protected member dim_feature_space
-	 * throws a shogun exception if dim<=0
-	 *
-	 */
-	void set_dim_feature_space(const int32_t dim);
-
-	/** computes new random coefficients IF test_rfinited() evaluates to false
-	 * test_rfinited() evaluates to TRUE if void set_randomcoefficients(...)
-	 * hase been called and the values set by set_dim_input_space(...) ,
-	 * set_dim_feature_space(...) and set_kernelwidth(...) are consistent to the
-	 * call of void set_randomcoefficients(...)
+	/** @brief Preprocessor that approximates Gaussian feature map.
 	 *
-	 * throws shogun exception if dim_feature_space <= 0 or dim_input_space <= 0
+	 * Gaussian kernel of form K(x,y) = exp(-||x - y||^2 / kernelwidth)
+	 * is approximated using the formula: 
 	 *
-	 * @return returns true if test_rfinited() evaluates to false and new
-	 * coefficients are computed
-	 * returns false if test_rfinited() evaluates to true and old random
-	 * coefficients are kept which were set by a previous call to void
-	 * set_randomcoefficients(...)
-	 *
-	 * this function is useful if you want to use apply_to_feature_vector but
-	 * cannot call before it fit(Features *f)
+	 * \f[
+	 * 	z(x) = \sqrt{2/D}\ cos(w'x + b)
+	 *  K(x, y) = \langle\phi(x),\phi(y)\rangle \approx z'(x)z(y)
```

Comment:
kernels are usually denoted by small "k"

Reply:
also the formula is not correct
k(x,y) \approx \phi(x) ^\top phi(y)
not sure what your z is you havent defined that

Reply:
Pretty sure formula is correct (taken directly from the paper), here's what it renders to.
![formula](https://user-images.githubusercontent.com/45007169/81812178-68b10300-9543-11ea-8d76-706d1aeb5215.png)
About z, this wouldn't suffice ?
![z](https://user-images.githubusercontent.com/45007169/81812632-0a385480-9544-11ea-8944-1a396435a2e1.png)

Reply:
ah sorry I misread it. I would remove the \phi and just do k(x,y) \approx z(x)^\top z(y)
I'd also remove the `'`